### ISSUE TITLE
'rsdical' is not returning results while 'radical' does

### ISSUE THREAD

**Issue by pramodthammaiah** [ankane/searchkick#534]
I have a product in my index with 'Radical' in its name. 

When I do the following searches, the product is returned:
'radical'
'redical'
'radicals'

However, when I search 'Rsdical' or 'rsdical' nothing is returned. I'd expect this query to return the product since the edit distance between 'rsdical'/'Rsdical' and 'Radical' is 1. 

Am I missing something? 

Here is the output from my rails server:
Product Search (6.4ms)  curl http://localhost:9200/products_development/_search?pretty -d '{"query":{"dis_max":{"queries":[{"match":{"_all":{"query":"Rsdical","operator":"and","boost":1,"fuzziness":1,"prefix_length":0,"max_expansions":3,"analyzer":"searchkick_search"}}},{"match":{"_all":{"query":"Rsdical","operator":"and","boost":1,"fuzziness":1,"prefix_length":0,"max_expansions":3,"analyzer":"searchkick_search2"}}},{"match":{"_all":{"query":"Rsdical","operator":"and","boost":10,"analyzer":"searchkick_search"}}},{"match":{"_all":{"query":"Rsdical","operator":"and","boost":10,"analyzer":"searchkick_search2"}}}]}},"size":100000,"from":0,"fields":[]}'
  Rendered searches/index.html.erb within layouts/application (0.7ms)

**Comment by ankane** [comment]
Two things to try:

1. `max_expansions` may be too low.  Try using the master branch and setting `misspellings: {max_expansions: 50}`.
2. See how Elasticsearch is tokenizing records with `radical` with the `default_index` analyzer, and how `Rsdical` is being tokenized with the `searchkick_search` and `searchkick_search2` analyzers.
https://github.com/ankane/searchkick#debugging-queries

**Comment by pramodthammaiah** [comment]
Thanks for the help!

Looks like something is going on with the stemmer:
```
>> Product.searchkick_index.tokens("Radical", analyzer: "default_index")
=> ["radic"]
>> Product.searchkick_index.tokens("rsdical", analyzer: "searchkick_search")
=> ["rsdical"]
>> Product.searchkick_index.tokens("rsdical", analyzer: "searchkick_search2")
=> ["rsdical"]
```
This seems like a corner case deep in Lucene's stemmer that's probably not straightforward to fix. Let me know if you have other ideas or just close if you agree :)

BTW, I loaded the gem from master, but setting `misspellings: {max_expansions: 50}` didn't change the behavior. 

-Pramod

ps. Huge thanks for developing this awesome gem. I just moved my project over from sunspot/solr and I'm blown away by how much easier everything is. The documentation is great and basic config just works (not the case with solr :smirk:)
